core/src/main/java/org/elasticsearch/cluster/metadata/SnapshotId.java

```diff
@@ -19,37 +19,78 @@
 
 package org.elasticsearch.cluster.metadata;
 
+import org.elasticsearch.common.UUIDs;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.snapshots.Snapshot;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
- * Snapshot ID - repository name + snapshot name
+ * SnapshotId - snapshot (repository name + snapshot name) + snapshot UUID
  */
-public class SnapshotId implements Streamable {
+public class SnapshotId implements Writeable {
 
-    private String repository;
+    public static final String UNASSIGNED_UUID = ""; // empty string so its compatible with old blob naming
```

Comment:
I wonder if we should use something like `"_na_"` similar to the `INDEX_UUID_NA_VALUE`. That is more an explicit value.